qcodes/utils/helpers.py

```diff
@@ -70,8 +72,16 @@ def default(self, obj):
             try:
                 s = super(NumpyJSONEncoder, self).default(obj)
             except TypeError:
-                # we cannot convert the object to JSON, just take a string
-                s = str(obj)
+                # See if the object supports the pickle protocol.
+                # If so, we should be able to use that to serialize.
+                if hasattr(obj, '__getnewargs__'):
```

Comment:
I wanted to make sure that this only gets called after default() as a fallback, so that it caused minimal changes to existing behavior. Thus, I needed to alternate if/elif and try/except nesting.